### PR TITLE
KIN-228 Webpack-gulp integration

### PR DESCRIPTION
**Pull request by Mauriciocr207**
[KIN-228](https://loopcrack.atlassian.net/browse/KIN-228) Large [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=loopcrack-org&utm_campaign=badge_info)

### Problem

Developers need an space where build all project resources. This to accelerate the development and generate assets to production.

### Solution

Reestructuring of gulp and tasks to compile SASS and use JS on project. Was created an integration made with webpack and gulp to minify, compile and get an standart CSS and JS files. 

### Requirements checklist

- [x] Compile SASS to minified and standart CSS for multiple files.
- [x] Bundle JS modules into one minified JS file for multiple files.
- [x] It is supported importing JS modules and SASS from node_modules.
- [x] Provide npm scripts to generate all JS and CSS to production.
- [x] Provide npm scripts to development mode in three areas: Admin, Public and Common.

### Testing

- [x] Run  the scripts provided on `package.json`. The scripts are described below.

- `npm run images`. Build all images in .webp, .avif, and jpg formats.
- `npm run dev:public`. Build public assets on dev mode
- `npm run dev:admin`. Build admin assets on dev mode
- `npm run dev:common`. Build common assets on dev mode
- `npm run dev`. Build public, admin and common assets on dev mode
- `npm run build`. Build all assets to production

[KIN-228]: https://loopcrack.atlassian.net/browse/KIN-228?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by CarlosJCD — APPROVED** [review]
looks good

**Review by CheloChidoone — APPROVED** [review]
The assets seem to be generated correctly, so it looks good for the objective of this PR.

**Review by LUGAMAFE — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -7,26 +7,40 @@
...
+    "node-waves": "^0.7.6",
     "postcss": "^8.4.27",
-    "sass": "^1.64.2"
+    "sass": "^1.64.2",
+    "simplebar": "^6.2.5",
+    "vinyl-named": "^1.1.0",
+    "webpack-stream": "^7.0.0"
+  },
+  "devDependencies": {
+    "gulp-sourcemaps": "^3.0.0"
```
[the remark this answers is not in the document]
Comment (Mauriciocr207) [reply package.json:44]: Yes, I was thinking about how dependencies could be devDependencies and I saw only one, the gulp-sourcemaps. Which other dependencies could be devDependencies?

**Review by LUGAMAFE — APPROVED** [review]
I only left a small nitpick. And please update the description of the tickets with the new commands

**Review by CarlosMay7 — APPROVED** [review]

**Review by Oswaldo-Chan — APPROVED** [review]

**Pull request merged**